spec/models/manageiq/providers/openshift/container_manager/refresh_parser_spec.rb

```diff
@@ -281,40 +281,99 @@ def parse_single_openshift_image_with_registry
   end
 
   describe "parse_build_pod" do
+    let (:basic_build_pod) do
+      {
+        :metadata => {
+          :name              => 'ruby-sample-build-1',
+          :uid               => 'af3d1a10-44c0-11e5-b186-0aaeec44370e',
+          :resourceVersion   => '165339',
+          :creationTimestamp => '2015-08-17T09:16:46Z',
+        },
+        :status   => {
+          :message                    => 'we come in peace',
+          :phase                      => 'set to stun',
+          :reason                     => 'this is a reason',
+          :duration                   => '33',
+          :completionTimestamp        => '50',
+          :startTimestamp             => '17',
+          :outputDockerImageReference => 'host:port/path/to/image',
+          :config                     => {
+            :name => 'ruby-sample-build',
+          },
+        }
+      }
+    end
+
+    let (:basic_build_config) do
+      {
+        :metadata => {
+          :name              => 'ruby-sample-build',
+          :uid               => 'af3d1a10-44c0-11e5-b186-0aaeec44370e',
+          :resourceVersion   => '165339',
+          :creationTimestamp => '2015-08-17T09:16:46Z',
+        },
+        :spec     => {
+          :serviceAccount            => 'service_account_name',
+          :completionDeadlineSeconds => '11',
+          :output                    => {
+            :to => {
+              :name => 'spec_output_to_name',
+            },
+          },
+          :source                    => {
+            :type => 'Git',
+            :git  => {
+              :uri => 'http://my/git/repo.git',
+            }
+          },
+        }
+      }
+    end
+
     it "handles simple data" do
+      build_pod = basic_build_pod.dup
```

Comment:
I think you want `.deep_dup` (everywhere you `.dup`), because you're mutating several levels inside.
Could also use `.deep_merge`.